packages/expansion-hub/src/start-keep-alive.ts

```diff
@@ -0,0 +1,16 @@
+import {ExpansionHubInternal} from "./internal/ExpansionHub";
+
+/**
+ * Starts a keep alive task for a given hub.
+ * @param hub hub to start the keep alive for
+ * @param interval interval in ms between each keep alive
+ */
+export function startKeepAlive(hub: ExpansionHubInternal, interval: number) {
+    if(hub.keepAliveTimer) {
+        //stop existing keep alive timer, so we can start a new one.
+        clearInterval(hub.keepAliveTimer);
+    }
+    hub.keepAliveTimer = setInterval(async () => {
+        await hub.sendKeepAlive();
+    }, interval);
```

Comment:
`setInterval()` won't do any error handling on the promise returned by the callback, so it shouldn't be an async function. Instead, we should do something like this:

```tyepscript
hub.sendKeepAlive()
	.catch ((e: any) => {
		// `RevHub` should probably be an `EventEmitter`, and we should send an `error` event at this point.
	});

Reply:
I figure that if a keepalive takes longer than interval (imagine a serial disconnection timeout), we want to prevent it from running again. I imagine that if the callback is async, setInterval should wait for it to complete before starting the next one, right?

Reply:
Fixed